openupgradelib/openupgrade.py

```diff
@@ -2606,19 +2606,80 @@ def delete_record_translations(cr, module, xml_ids):
         ),
     )
     for row in cr.fetchall():
-        query = """
-            DELETE FROM ir_translation
-            WHERE module = %s AND name LIKE %s AND res_id = %s;
-        """
-        logged_query(
-            cr,
-            query,
-            (
-                module,
-                row[0] + ",%",
-                row[1],
-            ),
-        )
+        model = row[0]
+        record_id = row[1]
+        if version_info[0] < 16:
+            query = """
+                DELETE FROM ir_translation
+                WHERE module = %s AND name LIKE %s AND res_id = %s;
+            """
+            logged_query(
+                cr,
+                query,
+                (
+                    module,
+                    model + ",%",
+                    record_id,
+                ),
+            )
+        else:
+            from psycopg2.extras import Json
+
+            table = model.replace(".", "_")
+            cr.execute(
+                """
+                SELECT isc.column_name
+                FROM information_schema.columns isc
+                JOIN ir_model_fields imf ON (
+                    imf.name = isc.column_name AND imf.model = %s)
+                JOIN ir_model_data imd ON (
+                    imd.model = 'ir.model.fields' AND imd.res_id = imf.id)
+                WHERE isc.table_name = % AND imd.module = %s
+                    AND imf.translate""",
+                (model, table, module),
+            )
+            list_columns = [x[0] for x in cr.fetchall()]
+            if not list_columns:
+                continue
+            columns = ", ".join(list_columns)
+            query = """
+                SELECT {columns}
+                FROM {table}
+                WHERE id = %s""".format(
+                table=table,
+                columns=columns,
+            )
+            cr.execute(
+                query,
+                (record_id,),
+            )
+            list_translations = [x[0] for x in cr.fetchall()]
+            new_translations = []
+            for translations in list_translations:
+                field_translations = translations or {}
+                if "en_US" in field_translations:
+                    clean_value = {"en_US": field_translations["en_US"]}
```

Comment:
Not sure if I should conserve the en_US translation or delete it too.

Reply:
I seams to me that it should be kept, because leaving a field with `{}` as a value seams to make it disappear in the interface.

I tried setting value `{}` to the name field of a res_country and after that I can no longer find it on the interface. Updating the `en_US` translation via the interface, did not bring back the translation for this res_country.

Reply:
Thanks!

Reply:
I think we should set `NULL`, not `{}` in order to avoid it disappearing from the interface. When Odoo detects the null value, it will load translations. (Not tested, but it's my two cents of my logic).

![Selection_1473](https://github.com/OCA/openupgradelib/assets/25005517/57871b0b-5707-40e1-9b8b-97e2a8d3a9a3)

Reply:
But this is usually called after `load_data` call, so in `en_US` we already have the proper updated value. A doubt I have right now is that that I think we are forcing or not `en_US` in such `load_data` method.

Reply:
> But this is usually called after load_data call, so in en_US we already have the proper updated value.

Are you sure?

> A doubt I have right now is that that I think we are forcing or not en_US in such load_data method.

We need to check.

Reply:
That's the purpose of the `load_data` method.